CDCgov/prime-reportstream
# no storing report file when response is 400

This PR gives consistent behavior relating to 400 responses when a bad submission is sent in. report_file records will no longer be saved if the response is a 400

Test Steps:
1. Run unit tests
2. Verify no errors

1. Submit an erroneous hl7 or csv file
2. verify no record was saved in report_file, but action_log is still saved

## Changes
- Move 'record received report' to after where errors are thrown during ingestion
- Update storing of action_log values to remove a reference to reportId if the report is not being stored
- Update tests to reflect the consistent logic

## Checklist

### Testing

## Linked Issues
- Fixes #6114
